server/zip.js

```diff
@@ -0,0 +1,83 @@
+const ApiKeys = require('../config/api-config');
+const client = require('twilio')(ApiKeys.twilioAccountSid, ApiKeys.twilioAuthToken);
+const twilio = require('twilio');
+const Promise = require('bluebird');
+const request = Promise.promisifyAll(require('request'), { multiArgs: true });
+
+const askForZipCode = (number, name) => {
+  client.messages.create({
+    to: number,
+    from: ApiKeys.twilioNumber,
+    body: 'Hey Gary, ' + 'text me your zip code and I can forecast the weather. Just like this: 94102. Give it a try :)'
+  }, (err, message) => {
+    if (err) {
+      console.log(err);
+    } else {
+      console.log(message.sid);
+    }
+  })
+}
+
+const getEachNum = (contacts, callback) => {
+  for (let name of Object.keys(contacts)) {
+    callback(contacts[name], name);
+  }
+}
+// getEachNum(ApiKeys.testContacts, sendSillyQnAToNums);
+getEachNum(ApiKeys.testContacts, askForZipCode);
+
+const convertToFahrenheit = function (degrees) {
+  return (degrees - 273) * 9 / 5 + 32;
+}
+
+const getWeatherForecast = (zipCode) => {
```

Comment:
Because you have promisified the request module on line 5, rather than returning a new promise, it will be better to return the request promise. Something like 

`return request.getAsync(....)
  .then(data => {
    // rest of your logic to accumulate message
    return message;
  })`

This way you can chain from the initial call in './server-config' and let the error handling propagate all the way through the Promise chain. This also removes the need to generate unnecessary promises.